plutus-metatheory/src/VerifiedCompilation/UForceDelay.lagda.md

```diff
@@ -115,6 +116,13 @@ data FD {X : Set} {{_ : DecEq X}} : Zipper X → X ⊢ → X ⊢ → Set₁ wher
   abs : FD (zipwk z) x x' → FD (z · y) (ƛ x) (ƛ x')
   last-delay : Translation (FD □) x x' → FD (force □) (delay x) x'
   last-abs : Translation (FD □) x x' → FD (□ · y) (ƛ x) (ƛ x')
+  ifThenElse :
+    Pure x'
+    → Pure y'
+    → Translation (FD □) b b'
+    → FD (force z) x x'
+    → FD (force z) y y'
+    → FD (force z) ((((force (builtin ifThenElse)) · b) · x) · y) ((((force (builtin ifThenElse)) · b') · x') · y')
```

Comment:
Hm. On the one hand, this is clever. On the other hand it looks like it's gonna involve a lot of backtracking. Not sure. Let's try it I guess.